src/test/pybind/test_cephfs.py

```diff
@@ -940,3 +940,178 @@ def test_multi_target_command():
     if isinstance(mds_status, list): # if multi target command result
         for mds_sessions in session_map:
             assert(list(mds_sessions.keys())[0].startswith('mds.'))
+
+
+class TestFcopyFile:
+    '''
+    Tests for fcopyfile() method of CephFS Python bindings.
+    '''
+
+    PERMS = 0o755
+
+    def get_perm_bits_from_stat_mode(self, mode):
+        mode_oct = oct(mode)
+        # get only last three digits, rest is not relevant
+        perm_bits_str = mode_oct[-3:]
+        # convert from str to octal literal value
+        perms = int(perm_bits_str, 8)
+
+        return perms
+
+    def test_with_dir(self, testdir):
+        '''
+        Test that fcopyfile() create a new dir with expected mode.
+        '''
+        cephfs.mkdir('dir1', self.PERMS)
+
+        cephfs.fcopyfile('dir1', 'dir2', self.PERMS)
+
+        stat_result = cephfs.stat('dir2')
+        perms = self.get_perm_bits_from_stat_mode(stat_result.st_mode)
+        assert perms == self.PERMS
+
+    def test_with_symlink(self, testdir):
+        '''
+        Test that fcopyfile() creates a new symlink with same size, mode and
+        pointing to the same file.
+        '''
+        fd = cephfs.open(b'file1', 'w', self.PERMS)
+        SIZE = 1000
+        cephfs.write(fd, b'1' * SIZE, 0)
+        cephfs.close(fd)
+        cephfs.symlink('file1', 'slink1')
+
+        cephfs.fcopyfile('slink1', 'slink2', self.PERMS)
+        stat_result = cephfs.stat('slink2')
+        assert stat_result.st_size == SIZE
+
+        perms = self.get_perm_bits_from_stat_mode(stat_result.st_mode)
+        assert perms == self.PERMS
+
+        lstat_result1 = cephfs.stat('slink1')
+        lstat_result2 = cephfs.stat('slink2')
+        assert lstat_result1.st_size == lstat_result2.st_size
+
+        path1 = cephfs.readlink('slink1', 4096)
+        path2 = cephfs.readlink('slink2', 4096)
+        assert path1 == path2
+
+    def test_with_regfile_of_size_1GB(self, testdir):
```

Comment:
These regfile size tests are all similar, could we break out most of the logic to a method?